tempodb/compactor.go

```diff
@@ -113,6 +118,17 @@ func (rw *readerWriter) doCompaction() {
 
 		// after a maintenance cycle bail out
 		if start.Add(rw.cfg.BlocklistPoll).Before(time.Now()) {
+			// count number of outstanding blocks remaining before next maintenance cycle
+			for {
+				leftToBeCompacted, _ := blockSelector.BlocksToCompact()
+				if len(leftToBeCompacted) == 0 {
+					break
+				}
+				for _, block := range leftToBeCompacted {
+					metricCompactionOutstandingBlocks.WithLabelValues(block.TenantID).Inc()
+				}
```

Comment:
We could:

Use counter.Add(len(leftToBeCompacted)) to remove this loop.

But:

If we're going to make this per tenant I'd prefer using a gauge as it would show is exactly how many blocks remaining per tenant.

Reply:
Changed to a gauge 👍